### PR TITLE
[qaucli] Add QLogic plugin support

### PR DESCRIPTION
**Pull request by vikas-goel**
Signed-off-by: Vikas Goel <vikas.goel@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/qaucli.py
Comment (TurboTurtle) [inline sos/report/plugins/qaucli.py:23]: Nack. Do not manually modify the `PATH` for this, the sos API already accounts for standardized `PATH`s, and additionally doing things this way can interfere with non-standard `--sysroot`s.

File: sos/report/plugins/qaucli.py
Comment (TurboTurtle) [inline sos/report/plugins/qaucli.py:28]: Ack.

File: sos/report/plugins/qaucli.py
Comment (TurboTurtle) [inline sos/report/plugins/qaucli.py:35]: Ack.

**Review by TurboTurtle — CHANGES_REQUESTED** [review]
Overall this looks good, just need to remove the `PATH` modifications.

The plugin name I think should be changed to just `qlogic`, but I can certainly be persuaded otherwise if there's a solid reason it shouldn't be `qlogic`.

File: sos/report/plugins/qaucli.py
Comment (vikas-goel) [reply sos/report/plugins/qaucli.py:23]: The CLI is `/opt/QLogic_Corporation/QConvergeConsoleCLI/qaucli`. What would be appropriate way to set the path?

**Comment by vikas-goel** [comment]
> Overall this looks good, just need to remove the `PATH` modifications.
> 
> The plugin name I think should be changed to just `qlogic`, but I can certainly be persuaded otherwise if there's a solid reason it shouldn't be `qlogic`.

There are other packages too to manage QLogic. For example, SANsurfer CLI (`scli`). Someone may want to develop a plugin for that. I can change the plugin name to `qlogic`. I think the filename should still be `qaucli.py`. What do you think?

File: sos/report/plugins/qaucli.py
Comment (TurboTurtle) [reply sos/report/plugins/qaucli.py:23]: You can just use that full path as the command.

If you wanted to make the resulting filenames more human friendly, you'd need to change your command loop a bit to something like:

```python
cmd = "/opt/QLogic_Corporation/QConvergeConsoleCLI/qaucli"
for subcmd in subcmds:
    self.add_cmd_output("%s %s" % (cmd, subcmd),
                        suggest_filename="quacli_%s" % subcmd)
```

Comment (vikas-goel) [reply sos/report/plugins/qaucli.py:23]: That won't work for options (`subcmd`) with space. Is there a possibility to change the sosreport framework to
1. pick command basename when creating output file,
2. create command alias as a function parameter, or
3. update PATH env variable?

Comment (vikas-goel) [reply sos/report/plugins/qaucli.py:23]: I just tried your suggestion. It worked for the space too. Thank you @TurboTurtle .

**Comment by vikas-goel** [comment]
> Overall this looks good, just need to remove the `PATH` modifications.
> 
> The plugin name I think should be changed to just `qlogic`, but I can certainly be persuaded otherwise if there's a solid reason it shouldn't be `qlogic`.

Done

**Comment by vikas-goel** [comment]
Hi @TurboTurtle ,
All suggestions incorporated here.

**Review by TurboTurtle — APPROVED** [review]
LGTM, just need a second ack from one of the other maintainers.

**Comment by vikas-goel** [comment]
Hi @TurboTurtle ,
How can we get another reviewer ack this and other similar PRs?

**Comment by TurboTurtle** [comment]
The other maintainers will look over and review as they are able - but I know at least 2 of them were on vacation this week. I imagine a second ack should be coming in the next week or so.

In any event, our next release is slated for mid-August, so these should have no issue being merged by that time.

**Comment by vikas-goel** [comment]
> In any event, our next release is slated for mid-August, so these should have no issue being merged by that time.

Any idea which Red Hat release will pick the mid-August sos release?

**Comment by vikas-goel** [comment]
> The other maintainers will look over and review as they are able - but I know at least 2 of them were on vacation this week. I imagine a second ack should be coming in the next week or so.
> 
> In any event, our next release is slated for mid-August, so these should have no issue being merged by that time.

Is there any update on when this can be merged?

File: sos/report/plugins/qaucli.py
Comment (pmoravec) [inline sos/report/plugins/qaucli.py:18]: Will the package name be matched on Debian/Ubuntu? Since we do case sensitive package name check and I see `qconvergeconsolecli` DEB package ("contrary" to `QConvergeConsoleCLI` RPM) in some manuals.

**Comment by pmoravec** [comment]
Up to the package name case sensitivity, ack.

File: sos/report/plugins/qaucli.py
Comment (vikas-goel) [reply sos/report/plugins/qaucli.py:18]: I don’t know about that. Worst case if the package name doesn’t match, it won’t execute. I don’t have access to that package nor do I know how on the disk the package extraction looks like. So, I don’t want to add the lowercase package name.

What’s your opinion?

Comment (pmoravec) [reply sos/report/plugins/qaucli.py:18]: > Worst case if the package name doesn’t match, it won’t execute.

Indeed. It isnt a bug, "just" possible lack of new functionality. Until @slashdd or @adam-stokes object (if they know `qconvergeconsolecli` DEB is really in lowercase), I am OK with merging it as is.

**Review by pmoravec — APPROVED** [review]

**Pull request merged**
